### ISSUE TITLE
Support `eachindex` for `ChunkSplitters.Enumerate`

### ISSUE THREAD

**Issue by carstenbauer** [JuliaFolds2/ChunkSplitters.jl#41]
I want this to work:

```julia
eachindex(enumerate(chunks(rand(10); n=5)))
```

Deeper reason is that we use `eachindex` in OhMyThreads.jl and the above is required for the following to work:

```julia
@tasks for (c, v) in enumerate(chunks(rand(10); n=5))
    # ...
end
```

As a simple solution, we could define

```julia
Base.keys(x::ChunkSplitters.Enumerate) = Base.OneTo(x.itr.n)
```

or, alternatively, `Base.eachindex` directly.

**Comment by lmiq** [comment]
I would go for `eachindex`, but any option is fine.

**Comment by lmiq** [comment]
I think this is better, because it works with `chunks(rand(10); size=2)` as well:

```
eachindex(ec::Enumerate{<:Chunk}) = Base.OneTo(length(ec.itr))
```

PR: https://github.com/JuliaFolds2/ChunkSplitters.jl/pull/42

**Comment by lmiq** [comment]
Released in 2.5.0 just now.
